doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+    ,1 /* minimum leaf size */, 1e-10 /* minimum gain */);
+// Print the MSE of the trained tree.
+std::cout << "MSE of trained tree is " << mse << "." << std::endl;
+
+// Compute prediction on the first test point.
+const double firstPrediction = rf.Predict(testData.col(0));
+std::cout << "Predicted value for first test point is " << firstPrediction
+    << "." << std::endl;
+
+// Compute predictions on test data.
+arma::rowvec testPredictions;
+rf.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "Average error on test set: " << testAverageError << "."
+    << std::endl;
+```
+
+---
+
+Load a random forest regressor and print some information about it.
+
+```c++
+mlpack::RandomForestRegressor rf;
+// This call assumes a random forest called "rf" has already been saved to
+// `rf.bin` with `data::Save()`.
+mlpack::data::Load("rf.bin", "rf", rf, true);
+
+std::cout << "The random forest in 'rf.bin' contains " << rf.NumTrees()
+    << " trees." << std::endl;
+if (rf.NumTrees() > 0)
+{
+  std::cout << "The first tree's root node has " << rf.Tree(0).NumChildren()
+      << " children." << std::endl;
+}
+```
+
+---
+
+Train a random forest regressor on categorical data, and compare its performance 
+with the performance of each individual tree:
+
+```c++
+// Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
+
+arma::rowvec responses;
+// See https://datasets.mlpack.org/telecom_churn.responses.csv.
+mlpack::data::Load("telecom_churn.responses.csv", responses, true);
+
+// Split data into training set (80%) and test set (20%).
+arma::mat trainData, testData;
+arma::rowvec trainResponses, testResponses;
+mlpack::data::Split(data, responses, trainData, testData, trainResponses,
+    testResponses, 0.2);
+
+// Create the tree.
+mlpack::RandomForestRegressor rf;
+// Train on the given dataset, specifying a minimum gain of 1e-10 and keeping the
+// default minimum leaf size.
+const double mse = rf.Train(trainData, info, trainResponses, 20 /*number of trees*/
+    ,1 /* minimum leaf size */, 1e-10 /* minimum gain */);
+
+// Compute test accuracy for each tree
+arma::rowvec testPredictions;
+for(size_t i = 0; i<rf.NumTrees(); i++)
+{
+  rf.Tree(i).Predict(testData, testPredictions);
+  const double testAverageError = arma::mean(testResponses - testPredictions);
+  std::cout << "Tree " << i << " has averageError: " << testAverageError << "."
+    << std::endl;
+}
+
+// Compute predictions on test data using whole forest.
+rf.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "The whole forest has Average error: " << testAverageError << "."
+    << std::endl;
+```
```

Comment:
Do you think you can either modify one of these examples or add another one that works on numeric-only data?  Ideally, in the examples, we demonstrate all (or virtually all) of the functionality described in the previous sections.